src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1325,6 +1327,52 @@ class spell_pri_prayer_of_mending_jump : public spell_pri_prayer_of_mending_Spel
     }
 };
 
+// 33110 - Prayer of Mending (Heal)
+class spell_pri_prayer_of_mending_heal : public spell_pri_prayer_of_mending_SpellScriptBase
+{
+    PrepareSpellScript(spell_pri_prayer_of_mending_heal);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_PRIEST_BENEDICTION,
+            SPELL_PRIEST_RENEW,
+            SPELL_PRIEST_DIVINE_SERVICE,
+            SPELL_PRIEST_PRAYER_OF_MENDING_AURA
+        });
+    }
+
+    void HandleEffectHitTarget(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        Unit* target = GetHitUnit();
+
+        if (AuraEffect const* divineService = caster->GetAuraEffect(SPELL_PRIEST_DIVINE_SERVICE, EFFECT_0))
+        {
+            if (Aura* prayerOfMending = target->GetAura(SPELL_PRIEST_PRAYER_OF_MENDING_AURA, caster->GetGUID()))
+            {
+                float healBonus = 1.0f + (divineService->GetAmount() * prayerOfMending->GetStackAmount() / 100.0f);
```

Comment:
Not sure if I'm supposed to obtain healingBonus from caster, effectVariance, then multiply the value for the bonus and lastly add healingBonusTaken from target first:

`
            float healingBonus = GetCaster()->SpellHealingBonusDone(GetHitUnit(), GetSpellInfo(), GetEffectValue(), HEAL, GetEffectInfo());
            float value = healingBonus + healingBonus * GetEffectVariance();
            value *= 1.0f + (divineService->GetAmount() * prayerOfMending->GetStackAmount() / 100.0f);
            value = GetHitUnit()->SpellHealingBonusTaken(GetCaster(), GetSpellInfo(), value, HEAL);
`